### PR TITLE
Swaps: Submitting button state

### PR DESCRIPTION
**Pull request by benisgold**
Fixes TEAM2-287

## What changed (plus any additional context for devs)
- added "Submitting" button state to swap button that activates after user initiates swap
- the original `handleSubmit` callback now returns early with a boolean that represents whether the swap was cancelled or not (if it was cancelled it means a warning modal is being displayed instead, and the swap button shouldn't change state yet). before returning this boolean, it begins executing the rest of the original `handleSubmit`, now in a new method called `submit`

## Screen recordings / screenshots

https://user-images.githubusercontent.com/15272675/180483904-a56bc0c6-bf58-4b30-82d6-adf86360bde9.mp4

## What to test
- swaps work
- button is enabled when it shouldn't be, or vice versa
- 
## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [ ] Did you test both iOS and Android?
- [ ] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [ ] If you added new critical path files, did you update the CODEOWNERS file?
- [ ] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/components/exchange/ConfirmExchangeButton.js [hunk 1]
```diff
@@ -122,9 +122,13 @@ export default function ConfirmExchangeButton({
   } else if (!isValidGas) {
     label = lang.t('button.confirm_exchange.invalid_fee');
   } else if (isSwapDetailsRoute) {
-    label = isHighPriceImpact
-      ? lang.t('button.confirm_exchange.swap_anyway')
-      : `${lang.t('button.confirm_exchange.swap')}`;
+    if (swapSubmitted) {
```
Comment (estebanmino) [inline src/components/exchange/ConfirmExchangeButton.js:125]: 🤝

File: src/screens/ExchangeModal.js [hunk 2]
```diff
@@ -517,6 +517,89 @@ export default function ExchangeModal({
...
 
+  const submit = useCallback(
+    async amountInUSD => {
+      setIsAuthorizing(true);
+      let NotificationManager = ios ? NativeModules.NotificationManager : null;
+      try {
+        const wallet = await loadWallet();
+        if (!wallet) {
+          setIsAuthorizing(false);
+          logger.sentry(`aborting ${type} due to missing wallet`);
```
Comment (estebanmino) [inline src/screens/ExchangeModal.js:528]: can set submitting to false if something happens with `loadWallet` and it didn't load? i manually made it fail and the button is stuck in `Submitting`

Comment (estebanmino) [reply src/screens/ExchangeModal.js:528]: probably similar to `setIsAuthorizing`

File: src/screens/ExchangeModal.js [hunk 3]
```diff
@@ -517,6 +517,89 @@ export default function ExchangeModal({
...
+          priceImpact: priceImpactPercentDisplay,
+          slippage: slippageInBips / 100,
+          type,
+        });
+        // Tell iOS we finished running a rap (for tracking purposes)
+        NotificationManager &&
+          NotificationManager.postNotification('rapCompleted');
+      } catch (error) {
+        setIsAuthorizing(false);
+        logger.log('[exchange - handle submit] error submitting swap', error);
```
Comment (estebanmino) [inline src/screens/ExchangeModal.js:575]: here as well, can we set submitting state to false if something happens with loadWallet and it didn't load? i manually made it fail and the button is stuck in Submitting

File: src/screens/ExchangeModal.js (hunk 2 above)
Comment (estebanmino) [reply src/screens/ExchangeModal.js:528]: nvm

File: src/screens/ExchangeModal.js (hunk 3 above)
Comment (estebanmino) [reply src/screens/ExchangeModal.js:575]: nvm

**Review by estebanmino — APPROVED** [review]
tried to break it and i couldn't 🪨

**Review by derHowie — APPROVED** [review]
lgtm, king

**Pull request merged**